### PR TITLE
[15017] External locators implementation

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

Adds implementation for the external locators feature.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- ❌ Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
- **N/A** New feature has been added to the `versions.md` file (if applicable). <-- Will be done on the PR to master
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation.   <-- Will be done on the PR to master

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
Just two little comments regarding some asserts.

**Comment by jsantiago-eProsima** [comment]
`ExternalLocatorsProcessorTests` are passing. CI failures unrelated.

File: src/cpp/rtps/network/ExternalLocatorsProcessor.cpp [hunk 1]
```diff
@@ -30,38 +30,293 @@ namespace fastdds {
...
+        const LocatorWithMask& local_locator,
+        const Locator& remote_locator)
+{
+    if (local_locator.kind == remote_locator.kind)
+    {
+        switch (local_locator.kind)
+        {
+            case LOCATOR_KIND_UDPv4:
+            case LOCATOR_KIND_TCPv4:
+                assert(32 >= local_locator.mask());
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/network/ExternalLocatorsProcessor.cpp:184]: Just read it from right to left. 😉

File: src/cpp/rtps/network/ExternalLocatorsProcessor.cpp [hunk 2]
```diff
@@ -30,38 +30,293 @@ namespace fastdds {
...
+                return address_matches(remote_locator.address + 12, local_locator.address + 12, local_locator.mask());
+
+            case LOCATOR_KIND_UDPv6:
+            case LOCATOR_KIND_TCPv6:
+            case LOCATOR_KIND_SHM:
+                assert(128 >= local_locator.mask());
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/network/ExternalLocatorsProcessor.cpp:190]: Just read it from right to left. 😉

**Review by jsantiago-eProsima — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Pull request merged**